File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/CardsPresenter.java
package com.dbottillo.mtgsearchfree.presenter;

import com.dbottillo.mtgsearchfree.model.Deck;
import com.dbottillo.mtgsearchfree.model.MTGCard;
import com.dbottillo.mtgsearchfree.model.MTGSet;
import com.dbottillo.mtgsearchfree.model.SearchParams;
import com.dbottillo.mtgsearchfree.view.CardsView;

public interface CardsPresenter extends BasicPresenter {

    void init(CardsView view);

    void loadCards(MTGSet set);

    void loadIdFavourites();

    void saveAsFavourite(MTGCard card, boolean reload);

    void removeFromFavourite(MTGCard card, boolean reload);

    void getLuckyCards(int howMany);

    void loadFavourites();

    void loadDeck(Deck deck);

    void doSearch(SearchParams searchParams);

    void loadCardTypePreference();

    void toggleCardTypeViewPreference();

}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/presenter/CardsPresenter.kt
package com.dbottillo.mtgsearchfree.presenter

import com.dbottillo.mtgsearchfree.resources.MTGCard
import com.dbottillo.mtgsearchfree.resources.MTGSet
import com.dbottillo.mtgsearchfree.view.CardsView

interface CardsPresenter : BasicPresenter {

    fun init(view: CardsView)

    fun loadCards(set: MTGSet)

    fun loadIdFavourites()

    fun saveAsFavourite(card: MTGCard)

    fun removeFromFavourite(card: MTGCard)

    fun getLuckyCards(howMany: Int)
}